app/channel.go

```diff
@@ -1330,8 +1321,18 @@ func (a *App) GetChannelUnread(channelId, userId string) (*model.ChannelUnread,
 }
 
 func (a *App) JoinChannel(channel *model.Channel, userId string) *model.AppError {
-	userChan := a.Srv.Store.User().Get(userId)
-	memberChan := a.Srv.Store.Channel().GetMember(channel.Id, userId)
+	userChan := make(chan store.StoreResult, 1)
```

Comment:
Wait group?

Reply:
Wait group + shared variables?

Reply:
I'm ok with wait groups, is a different approach and have advantages like the possibility of use directly variables without sending it through the channel and without dealing with store.StoreResult at all, but I think will be more code and more more error prone. The channel approach is systematic and simple.

Reply:
This should be something like:
```go
      var user *model.User
      var userErr *model.AppError
      var member *model.ChannelMember
      var memberErr *model.AppError
      var wg sync.WaitGroup
      wg.Add(2)  
      go func() {
          defer wg.Done()
          user, userErr = a.Srv.Store.User().Get(userId) 
      }()
      go func() {
          defer wg.Done()
          member, memberErr = a.Srv.Store.Channel().GetMember(channel.Id, userId) 
      }()      
      wg.Wait()
      if userErr != nil {
          return nil, userErr
      }
      if memberErr != nil {
          return nil, memberErr
      }
```

For me looks like less clear in general. But is an opinion

Reply:
See https://community-daily.mattermost.com/core/pl/eitu1bjhufnpffn3m83836759e